Dockerfile

```diff
@@ -1,29 +1,55 @@
-FROM golang:1.13 as secrets-provider-builder
+# =================== BASE BUILD LAYER ===================
+# this layer is used to prepare a common layer for both debug and release builds
+FROM golang:1.13 as secrets-provider-builder-base
 MAINTAINER CyberArk Software Ltd.
 
 ENV GOOS=linux \
     GOARCH=amd64 \
     CGO_ENABLED=0
 
-COPY . /opt/secrets-provider-for-k8s
+RUN apt-get update
 
-WORKDIR /opt/secrets-provider-for-k8s
+RUN go get -u github.com/jstemmer/go-junit-report && \
+    go get github.com/smartystreets/goconvey
+
+WORKDIR /opt/secrets-provider
 
 EXPOSE 8080
 
-RUN apt-get update && apt-get install -y jq
+COPY go.mod .
+COPY go.sum .
 
-RUN go get -u github.com/jstemmer/go-junit-report && \
-    go get github.com/smartystreets/goconvey
+# Add a layer of prefetched modules so in case of rebuild the modules are already cached
+RUN go mod download
+
+# =================== RELEASE BUILD LAYER ===================
+# this layer is used to build the release binaries
+FROM secrets-provider-builder-base as secrets-provider-builder
+
+COPY . .
 
 RUN go build -a -installsuffix cgo -o secrets-provider ./cmd/secrets-provider
 
+# =================== DEBUG BUILD LAYER ===================
+# this layer is used to build the debug binaries
+FROM secrets-provider-builder-base as secrets-provider-builder-debug
+
+# Build Delve
+RUN go get github.com/go-delve/delve/cmd/dlv
+
+EXPOSE 40000
+
+COPY . .
+
+RUN go build -a -installsuffix cgo -gcflags="all=-N -l" -o secrets-provider ./cmd/secrets-provider
+
 # =================== BUSYBOX LAYER ===================
 # this layer is used to get binaries into the main container
 FROM busybox
 
-# =================== MAIN CONTAINER ===================
-FROM alpine:3.11 as secrets-provider
+# =================== BASE MAIN CONTAINER ===================
+# this layer is used to prepare a common layer for both debug and release containers
```

Comment:
can you please add a blank line after this comment. Or maybe move this comment to be above the `BASE MAIN CONTAINER` line.

Reply:
We don't do it for any other layer comment. Why should I do it here?

Reply:
you are correct, i didn't notice that was the pattern.